integration/src/test/scala/org/knora/webapi/e2e/admin/ProjectsADME2ESpec.scala

```diff
@@ -125,11 +119,14 @@ class ProjectsADME2ESpec extends E2ESpec with ProjectsADMJsonProtocol {
           BasicHttpCredentials(rootEmail, testPass),
         )
         val response: HttpResponse = singleAwaitingRequest(request)
-        logger.debug(s"response: {}", response)
         assert(response.status === StatusCodes.OK)
 
-        val settings: ProjectRestrictedViewSettingsADM =
-          AkkaHttpUtils.httpResponseToJson(response).fields("settings").convertTo[ProjectRestrictedViewSettingsADM]
+        val settings =
+          ProjectRestrictedViewSettingsGetResponseADM.codec
+            .decodeJson(responseToString(response))
+            .getOrElse(throw new AssertionError("Could not decode response"))
```

Comment:
Three things:
1. Is possible to avoid throwing here or rather not?
2. Could this be unified? In other places you have used `"JSON decoding failed"`.
3. You are missing dot at the end of both kind of messages in all instances ;)

Reply:
1. it's test code, so I'm not really bothered by the throwing; but yes, it could be avoided, it would just make the test slightly more verbose. Would you prefer me to change it?
2. sure
3. right

Reply:
Thanks, all good. I omitted the fact this is in tests package...